### PR TITLE
CEDS 1059 - Associate a single DUCR with a new or existing MUC

### PR DESCRIPTION
**Pull request by benjaminparker**

### CODE REVIEW

File: app/assets/javascripts/show-hide-content.js [hunk 1]
```diff
@@ -13,8 +13,8 @@
     // Radio and Checkbox selectors
     var selectors = {
       namespace: 'ShowHideContent',
-      radio: '.block-label[data-target] input[type="radio"]',
-      checkbox: '.block-label[data-target] input[type="checkbox"]'
+      radio: '.multiple-choice[data-target] input[type="radio"]',
+      checkbox: '.multiple-choice[data-target] input[type="checkbox"]'
```
Comment (ed0906) [inline app/assets/javascripts/show-hide-content.js]: This fixes the conditional reveal radio button JS

File: test/base/ViewMatchers.scala [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.jsoup.nodes.Element
+import org.jsoup.select.Elements
+import org.scalatest.matchers._
+
+//noinspection ScalaStyle
+trait ViewMatchers {
```
Comment (ed0906) [inline test/base/ViewMatchers.scala]: I've pulled this in from Tariffs project as the Matchers are useful and allow us to JSOUP based tests using the standard Scala test style e.g. `element must haveAttribute("href", "http://abc.com")`

File: app/controllers/AssociateDucrConfirmationController.scala [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+class AssociateDucrConfirmationController @Inject()(
+  authenticate: AuthAction,
+  journeyType: JourneyAction,
+  errorHandler: ErrorHandler,
+  mcc: MessagesControllerComponents,
+  associateDucrConfirmPage: associate_ducr_confirmation
+)(implicit appConfig: AppConfig)
+    extends FrontendController(mcc) with I18nSupport {
+
+  def displayPage(): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
```
Comment (PauloPortugal) [inline app/controllers/AssociateDucrConfirmationController.scala]: :+1:

File: app/controllers/AssociateDucrController.scala [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.CustomsCacheService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import views.html.associate_ducr
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+@Singleton
+class AssociateDucrController @Inject()(
```
Comment (PauloPortugal) [inline app/controllers/AssociateDucrController.scala]: Awesome controller! Clean, short and sweet

File: app/controllers/AssociateDucrSummaryController.scala [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+      a: Option[AssociateDucr] <- cacheService.fetchAndGetEntry[AssociateDucr](movementCacheId(), AssociateDucr.formId)
+      associateDucr = a.getOrElse(throw IncompleteApplication)
+    } yield Ok(associateDucrSummaryPage(associateDucr, mucrOptions.mucr))
+  }
+
+  def submit(): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
+    for {
+      m: Option[MucrOptions] <- cacheService.fetchAndGetEntry[MucrOptions](movementCacheId(), MucrOptions.formId)
+      mucrOptions = m.getOrElse(throw IncompleteApplication)
```
Comment (PauloPortugal) [inline app/controllers/AssociateDucrSummaryController.scala]: Interesting... This is where Cats, the (in)famous library works well. ;)

File: app/handlers/ErrorHandler.scala [hunk 6]
```diff
@@ -47,6 +48,8 @@ class ErrorHandler @Inject()(appConfig: AppConfig, val messagesApi: MessagesApi,
       case _: NoActiveSession => toGGLogin(rh.uri)
       case _: InsufficientEnrolments =>
         Results.SeeOther(routes.UnauthorisedController.onPageLoad().url)
+      case _: IncompleteApplication =>
```
Comment (PauloPortugal) [inline app/handlers/ErrorHandler.scala]: :+1:

File: app/controllers/AssociateDucrSummaryController.scala (hunk 5 above)
Comment (ed0906) [reply app/controllers/AssociateDucrSummaryController.scala]: Lets not bring that up again :p

**Pull request merged**